google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/javadsl/GooglePubSub.scala

```diff
@@ -16,43 +16,76 @@ import scala.collection.JavaConverters._
 import scala.compat.java8.FutureConverters._
 import scala.concurrent.Future
 
+/**
+ * Java DSL for Google Pub/Sub
+ */
 object GooglePubSub {
 
+  /**
+   * Creates a flow to that publishes messages to a topic and emits the message ids.
+   */
   def publish(topic: String,
               config: PubSubConfig,
               parallelism: Int): Flow[PublishRequest, java.util.List[String], NotUsed] =
     GPubSub
-      .publish(topic = topic, config = config, parallelism = parallelism)
+      .publish(topic, config, parallelism)
       .map(response => response.asJava)
       .asJava
 
+  /**
+   * Creates a flow to that publishes messages to a topic and emits the message ids and carries a context
+   * through.
+   */
   def publishWithContext[C](topic: String,
                             config: PubSubConfig,
                             parallelism: Int): FlowWithContext[PublishRequest, C, java.util.List[String], C, NotUsed] =
     GPubSub
-      .publishWithContext[C](topic = topic, config = config, parallelism = parallelism)
+      .publishWithContext[C](topic, config, parallelism)
       .map(response => response.asJava)
       .asJava
 
+  /**
+   * Creates a source pulling messages from a subscription.
+   */
   def subscribe(subscription: String, config: PubSubConfig): Source[ReceivedMessage, Cancellable] =
     GPubSub
-      .subscribe(subscription = subscription, config = config)
+      .subscribe(subscription, config)
       .asJava
 
+  /**
+   * Creates a flow pulling messages from a subscription.
+   */
   def subscribeFlow(subscription: String, config: PubSubConfig): Flow[Done, ReceivedMessage, Future[NotUsed]] =
     GPubSub
-      .subscribeFlow(subscription = subscription, config = config)
+      .subscribeFlow(subscription, config)
       .asJava
 
+  /**
+   * Creates a sink for acknowledging messages on a subscription.
+   *
+   * @deprecated Use `acknowledge` without `parallelism` param (since 2.0.0)
+   */
+  @Deprecated
```

Comment:
Is this redundant?

Reply:
Most likely, I don't remember how much deprecation one should add to mark it deprecated for Java users...